### PR TITLE
allow compressed raw files

### PR DESCRIPTION
**Pull request by schmucr1**
Dear @ajulienla  and @hatjek 
I modified the code so that one can read also gzip compressed raw data, e.g. matrix.mtx.gz
The code is not very elegant, I am sure one can improve there.
Could you please have a check and let me know your suggestions?
Thanks,
R.

### CODE REVIEW

File: besca/Import/_read.py [hunk 1]
```diff
@@ -116,16 +126,26 @@ def read_mtx(
...
-    adata = read(os.path.join(filepath, 'matrix.mtx'), cache= True).T  #transpose the data
+    if (gzfiles == 'gz'):
+      adata = read(os.path.join(filepath, 'matrix.mtx.gz'), cache= True).T  #transpose the data
+    else:
+      adata = read(os.path.join(filepath, 'matrix.mtx'), cache= True).T  #transpose the data
 
     print('adding cell barcodes')
-    adata.obs_names = pd.read_csv(os.path.join(filepath, 'barcodes.tsv'), header=None)[0]
-
+    if (gzfiles == 'gz'):
```
Comment (ajulienla) [inline besca/Import/_read.py:139]: Could we group all reading under one if condition
```
 if gzfiles == 'gz :
```
?

File: besca/Import/_read.py [hunk 2]
```diff
@@ -116,16 +126,26 @@ def read_mtx(
     returns an AnnData object
     """
 
-    assert_filepath(filepath)
+    gzfiles = assert_filepath(filepath)
 
+# NEW check whether gzipped version of matrix.mtx exists and if yes, then read the gzipped version otherwise the unzipped version
```
Comment (ajulienla) [inline besca/Import/_read.py:131]: I would indent comment to match code

**Review by ajulienla — APPROVED** [review]
Hi,
Tested and working. 
I approve as is.
As you wanted some feedback, I would group ** if ** in the read part.

**Comment by schmucr1** [comment]
Thanks @ajulienla , please polish all my code as you think is best!
Remove unnecessary comments and correct the `if`
Merci!

**Pull request merged**